kuka_kr3_support/urdf/kr3r540_macro.xacro

```diff
@@ -0,0 +1,171 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://wiki.ros.org/xacro">
+  <xacro:include filename="$(find kuka_resources)/urdf/common_constants.xacro"/>
+  <xacro:include filename="$(find kuka_resources)/urdf/common_materials.xacro"/>
+
+  <xacro:macro name="kuka_kr3r540" params="prefix">
+    <link name="${prefix}base_link">
+      <visual>
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <geometry>
+          <mesh filename="package://kuka_kr3_support/meshes/kr3r540/visual/base_link.dae" />
+        </geometry>
+        <xacro:material_kuka_pedestal />
+      </visual>
+      <collision>
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <geometry>
+          <mesh filename="package://kuka_kr3_support/meshes/kr3r540/collision/base_link.stl" />
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_1">
+      <visual>
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <geometry>
+          <mesh filename="package://kuka_kr3_support/meshes/kr3r540/visual/link_1.dae" />
+        </geometry>
+        <xacro:material_kuka_ral_pure_white />
+      </visual>
+      <collision>
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <geometry>
+          <mesh filename="package://kuka_kr3_support/meshes/kr3r540/collision/link_1.stl" />
+        </geometry>
+      </collision>
+    </link>
+    <link name="${prefix}link_2">
+      <visual>
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <geometry>
+          <mesh filename="package://kuka_kr3_support/meshes/kr3r540/visual/link_2.dae" />
```

Comment:
Using Collada here is nice, but if we're not going to texture/colour specific parts, but use solid colours instead, then they are essentially ascii versions of STLs.

I'm not against merging this, but a quick test shows that the STL version of `link_2.dae` is only `341 KB`, versus `1.1 MB` for the Collada file.

Reply:
I feel like this applies to many of the existing models?  Do the support packages using Collada actually use textures?

Reply:
Not textures, but they do typically have coloured vertex groups. That is not possible with STLs, hence the Collada files.